### PR TITLE
Fix read msg

### PR DESCRIPTION
**Pull request by oliv3r**
This PR is based upon #88 so please merge that one first :)

This PR fixes a small bug in readMsg where zero messages are not ignored in the rec file resulting in not being able to reproduce files.

### CODE REVIEW

File: se/msg.py [hunk 1]
```diff
@@ -114,8 +114,11 @@ def readMsg(inFile, recFile, mode):
             else:
                 msg += nextByte
         msg = msg[:-magicLen]  # strip the magic number from the end
-    if len(msg) > 0:  # don't log zero length messages
-        logger.message("-->", dataInSeq, magic + msg, inFile.name)
+
+    if len(msg) < 1:  # don't log zero length messages
```
Comment (ericbuehl) [inline se/msg.py]: good catch :)

However, don't you just want to just change the condition?  By adding the return you are changing what data is recorded.  Is this desirable?  cc @jbuehl 

That said, a more pythonic test would be something like 
```
if msg:
    logger.message("-->", dataInSeq, magic + msg, inFile.name)
```
https://www.python.org/dev/peps/pep-0008/#programming-recommendations

File: test/test.sh [hunk 2]
```diff
@@ -1,30 +1,71 @@
-#!/bin/bash -e
```
Comment (ericbuehl) [inline test/test.sh]: you can remove these changes from this PR.  since they don't overlap we can merge them independently

File: se/msg.py (hunk 1 above)
Comment (oliv3r) [reply se/msg.py]: actually, by not adding the return we are already changing the data recorded, as invalid messages get now get added to the log as an empty message.
e.g. if the message contains only half a message (e.g. a pcap file that starts midway tranmission). Our current code discards this message as the CRC does not match. We now have a zero length message which does get added to the rec log. This is not present in the json and technically was not present in the source.

As for the python style; the original code was explicitly checking for the length, so I stuck with that. If @jbuehl can agree that the existence of a message is good enough; I'll change it as part of this PR.

File: test/test.sh (hunk 2 above)
Comment (oliv3r) [reply test/test.sh]: yeah i can rebase them on master

File: se/msg.py (hunk 1 above)
Comment (oliv3r) [reply se/msg.py]: right now, for me, i use it to log all data that goes through semonitor, as the output json is not always accurate. For example. Message XYZ is currently unknown, so gets discarded or garbled in the log (timestamps in the event log for example), but does get logged in the rec file. If we now properly decode these logs however, I cannot go and backtrack my json logs. With the rec logs however I can do exactly this.

Comment (oliv3r) [reply se/msg.py]: as for the zero length message, right now, (and this is from 2 month old memories) if i have a rec file and i run it through semonitor again (which, with the rec option should produce the exact same output) it currently does not, but instead adds a new zero length message to the start. This zero length message was not there before, but it is now and thus the rec files (in and out) do no longer match.

I think if you run the new test case attached to this issue, with the old data, you'll see what i mean.

Again, this happens due to half message starts. With the replay (rec in -> rec out; cmp) i have not noticed any difference after this change however.

Comment (jbuehl) [reply se/msg.py]: Assuming you are not using an active mode @oliv3r and you want an exact copy of the input file, why not put `tee` in the pipe before semonitor.py to send it to a file?  The record file would still be useful for recording both sides of the conversation in active mode.

Comment (oliv3r) [reply se/msg.py]: sadly, your assumptions are invalid :p i'm using master/active rs-485 mode :)

but i'm thinking of cooking up a simple daemon that does the tcp -> rs-485 step (and decodes the packet (not the payload) allowing for multiple users (semonitor, other tools) to talk to the inverter simultaneously.

**Comment by jbuehl** [comment]
I created a new PR #105 to address this issue.

**Comment by oliv3r** [comment]
I changed this PR to only include the additional tests, as #50 solves this much better.

**Comment by ericbuehl** [comment]
@oliv3r can you rebase your branch on jbuehl/master? for some reason github is still showing the additional commits that are already in master

**Comment by oliv3r** [comment]
I rebased on @jbuehl branch but as you do squash merge, that's not accurate at that time anymore. I rebased onto master so should be okay now.

**Pull request merged**